pkg/backup/backup.go

```diff
@@ -67,7 +67,17 @@ func (ac ActionContext) log(msg string, args ...interface{}) {
 type Action interface {
 	// Execute is invoked on an item being backed up. If an error is returned, the Backup is marked as
 	// failed.
-	Execute(ctx ActionContext, item map[string]interface{}, backup *api.Backup) error
+	Execute(ctx ActionContext, item map[string]interface{}, backup *api.Backup) ([]itemKey, error)
```

Comment:
How about instead of returning the keys of additional items to back up, we just let the action call back to the itemBackupper to backup the additional items? i.e. each action implementation holds a reference to an itemBackupper. We'd need to make sure that the itemBackupper could look up the action instead of having that be part of kubernetesBackupper#backupResource. This way, each item can immediately back up transitive dependencies and add them to the backupContext's set of already backed up items.

Reply:
I like it in theory; a bunch of stuff will have to move around because we'll want to check resource includes/excludes, label-selector, backed up items before backing up the PV and this is all done outside the backupItem method now.